src/utils/spawnAsync.ts

```diff
@@ -56,15 +56,16 @@ export async function spawnAsync(
             } else if (code) {
                 let errorMessage = localize('vscode-docker.utils.spawn.exited', 'Process \'{0}\' exited with code {1}', command.length > 50 ? `${command.substring(0, 50)}...` : command, code);
 
-                if (!onStderr && stderrBuffer) {
-                    // If there was no progress handler for stderr, then the error output would be lost, so help mitigate that by putting the error output into the Error we throw
+                if (stderrBuffer) {
                     errorMessage += localize('vscode-docker.utils.spawn.exitedError', '\nError: {0}', bufferToString(stderrBuffer));
                 }
 
                 const error = <ExecError>new Error(errorMessage);
 
                 error.code = code;
                 error.signal = signal;
+                error.stdErrHandled = onStderr != null;

```

Comment:
If we are going with this approach, why consider only onStderr and ignoring stderrBuffer.

Reply:
I don't see that it adds any safety to include stdErrBuffer in this flag, since technically it is optional. The flag's purpose is only regarding the existence of the handler.

Reply:
We have to ignore stderrBuffer, because if the process exits with a non-zero code we reject rather than return (i.e. we throw an exception). The caller will have never gotten the stderr result.